### PR TITLE
Automatically converted to MSL 4.

### PR DESCRIPTION
**Pull request by HansOlsson**
To stay compatible.
Issues to do:

- [x] Check result of conversion.
- [x] Move script?
- [x] Possibly decide on a different version number.

### CODE REVIEW

**Comment by HansOlsson** [comment]
> Would it not make sense to use a conversion script here?

Yes, and it would probably make sense in general to see which functions in M_LS2 could directly be replaced by MSL-variants.

My primary goal was just to ensure that it was usable without needlessly relying on ObsoleteModelica4.

**Comment by HansOlsson** [comment]
Oh, missed the comment by @beutlich 

To clarify:
1. There is a conversion to ObsoleteModelica4 for the used function so there is no issue with MSL-conversions, but it relies on ObsoleteModelica4.
2. The ObsoleteModelica4 function says that you should use a new Lapack-function instead, I first did that.
3. Then realized that MSL had an almost identical wrapper, and then I called that wrapper.
4. Thus for conversion I mean that instead of having functions in M_LS2 that are almost just copies of MSL-functions it would make sense to convert those M_LS2 functions to the MSL counterparts. That is not something we could (easily) do automatically.

File: Modelica_LinearSystems2/Internal/frequencyVector2.mo [hunk 1]
```diff
@@ -32,8 +32,8 @@ algorithm
   // Determine f_min2, f_max2 (auto or manual)
   if autoRange then
     (w_min,w_max) := Internal.frequencyRangeBode2(Zeros, Poles);
-    f_min2 := SI.Conversions.to_Hz(w_min);
-    f_max2 := SI.Conversions.to_Hz(w_max);
+    f_min2 := Modelica.Units.Conversions.to_Hz(w_min);
```
Comment (beutlich) [inline Modelica_LinearSystems2/Internal/frequencyVector2.mo]: Why was this second step necessary? Is this a tool issue or a problematic conversion in MSL 4?

Comment (HansOlsson) [reply Modelica_LinearSystems2/Internal/frequencyVector2.mo]: A tool issue, probably due to changes in development version. 
I will investigate it later, but I first needed this version.

**Comment by HansOlsson** [comment]
After some tool-improvements, the only non-automatic steps would be:

- Correcting the conversion information for M_LS2-versions _before_ 2.3.5, which I believe is incorrect in M_LS2 master. Currently the annotations states that 2.3.5 requires a conversion but 2.3.0 doesn't, which is odd (although possible). I assume both should use the same conversion - as in this PR.
- Replacing the functions calling ObsoleteModelica4 functions with similar ones in MSL. In general I believe we should reconsider those functions duplicating MSL-ones, but that is a M_LS2 specific development.

**Comment by HansOlsson** [comment]
When checking the result I noticed that WIP has some broken classes, as I understand that is not a regression (but it would be good to clean up separately). Thus the check is excluding that sub-package

**Comment by HansOlsson** [comment]
It should now be ready. The _possibly_ remaining issues before releasing it are:

- Deciding what version-number to use.
- Additional improvements unrelated to MSL (remove WIP?)
- Use MSL-functions directly instead of copied functions (to ease maintenance)

File: Modelica_LinearSystems2/Math/Matrices/Internal/QR.mo [hunk 2]
```diff
@@ -18,7 +18,7 @@ algorithm
      + "] has more columns as rows.
 This is not allowed when calling Modelica.Matrices.QR(A).");
```
Comment (beutlich) [inline Modelica_LinearSystems2/Math/Matrices/Internal/QR.mo]:
```suggestion
This is not allowed when calling Modelica.Math.Matrices.QR(A).");

```

Was also wrong in copied function Modelica.Math.Matrices.QR and fixed there by https://github.com/modelica/ModelicaStandardLibrary/commit/412d17023bd7613ef226a611271f5001cbb54e78.

**Comment by tobolar** [comment]
@HansOlsson 
> Correcting the conversion information for M_LS2-versions before 2.3.5, which I believe is incorrect in M_LS2 master. Currently the annotations states that 2.3.5 requires a conversion but 2.3.0 doesn't, which is odd (although possible). I assume both should use the same conversion - as in this PR.

I guess the only conversion ever done is that in https://github.com/modelica/Modelica_LinearSystems2/blob/master/Modelica_LinearSystems2/Resources/Scripts/Dymola/ConvertLinearSystems2_from_2.3.4.mos which reflects significant changes done back in 2017. 
So maybe this is rather a problem of unsifficiently chosen version number.

**Comment by tobolar** [comment]
> The possibly remaining issues before releasing it are:
> - Deciding what version-number to use.
> - Additional improvements unrelated to MSL (remove WIP?)
> - Use MSL-functions directly instead of copied functions (to ease maintenance)

The last two points depend on the time line for releasing. IMO LS2 has not been pushed forward that much in the last time. 
So for me it is fine to release a MSL 4.0.0 compatible version shortly and address those two points (and possibly more) afterwards in the next months.

File: Modelica_LinearSystems2/Controller/Templates/SimpleObserverStateSpaceControl.mo [hunk 3]
```diff
@@ -9,7 +9,8 @@ partial model SimpleObserverStateSpaceControl
   Modelica.Blocks.Math.Feedback feedback[feedbackMatrix.nout]
     annotation (Placement(transformation(extent={{-30,-10},{-10,10}})));
   replaceable Internal.PlantTemplate plant(n=
-        feedbackMatrix.nin, m=feedbackMatrix.nout) constrainedby Internal.PlantTemplate
+        feedbackMatrix.nin, m=feedbackMatrix.nout) constrainedby
+    Internal.PlantTemplate
```
Comment (tobolar) [inline Modelica_LinearSystems2/Controller/Templates/SimpleObserverStateSpaceControl.mo:13]: I guess this change brakes a good readability of the code.  Could you revert this? (Ditto in the following files/classes.)

File: Modelica_LinearSystems2/Examples/ZerosAndPoles/plotBodeFilter2.mo [hunk 4]
```diff
@@ -7,11 +7,12 @@ function plotBodeFilter2 "Show low pass filters of all filter types"
...
 
   input Integer order(min=1) = 4 "Order of filter";
-  input Modelica.SIunits.Frequency f_cut=1 "Cut-off frequency";
+  input Modelica.Units.SI.Frequency f_cut=1 "Cut-off frequency";
   input Real A_ripple(unit="dB") = 3
     "Pass band ripple for Chebyshev filter (otherwise not used)";
-  input Modelica.SIunits.Frequency f_min(min=0) = 0.1 "Minimum frequency value";
-  input Modelica.SIunits.Frequency f_max(min=0) = 10 "Maximum frequency value";
+  input Modelica.Units.SI.Frequency f_min(min=0) = 0.1
+    "Minimum frequency value";
```
Comment (tobolar) [inline Modelica_LinearSystems2/Examples/ZerosAndPoles/plotBodeFilter2.mo:14]: Revert line break here.

File: Modelica_LinearSystems2/Examples/ZerosAndPoles/plotBodeFilter3.mo [hunk 5]
```diff
@@ -7,11 +7,12 @@ function plotBodeFilter3 "Show high pass filters of all filter types"
...
 
   input Integer order(min=1) = 4 "Order of filter";
-  input Modelica.SIunits.Frequency f_cut=1 "Cut-off frequency";
+  input Modelica.Units.SI.Frequency f_cut=1 "Cut-off frequency";
   input Real A_ripple(unit="dB") = 3
     "Pass band ripple for Chebyshev filter (otherwise not used)";
-  input Modelica.SIunits.Frequency f_min(min=0) = 0.1 "Minimum frequency value";
-  input Modelica.SIunits.Frequency f_max(min=0) = 10 "Maximum frequency value";
+  input Modelica.Units.SI.Frequency f_min(min=0) = 0.1
+    "Minimum frequency value";
```
Comment (tobolar) [inline Modelica_LinearSystems2/Examples/ZerosAndPoles/plotBodeFilter3.mo:14]: Revert line break here.

Comment (HansOlsson) [reply Modelica_LinearSystems2/Examples/ZerosAndPoles/plotBodeFilter3.mo]: That should now be corrected.

File: Modelica_LinearSystems2/Examples/ZerosAndPoles/plotBodeFilter2.mo (hunk 4 above)
Comment (HansOlsson) [reply Modelica_LinearSystems2/Examples/ZerosAndPoles/plotBodeFilter2.mo]: That should now be corrected.

File: Modelica_LinearSystems2/Controller/Templates/SimpleObserverStateSpaceControl.mo (hunk 3 above)
Comment (HansOlsson) [reply Modelica_LinearSystems2/Controller/Templates/SimpleObserverStateSpaceControl.mo]: That should now be corrected.

**Comment by HansOlsson** [comment]
> Your automatic conversion often deletes blanks around equal sign, i.e. " = " into "=". This shall be omited in the future.

I tried to revert all of them, and also unnecessary spaces in `import     Modelica.Units.SI;`
I believe one of the causes of this is that the new type-name is slightly longer than the earlier one.

I noticed that it was not entirely consistent earlier, so I only tried to restore it how it was - not make it consistent.

> The new library relase shall be prepared in a separate PR.

Ok, I changed it to just have it marked as "2.3.6 Development", and we can later decide on the actual release name (whether it is 2.3.6 or 2.4.0 or something else).

The reason is just that if I don't want a mix of two 2.3.5 versions using different MSL-versions.

File: Modelica_LinearSystems2/Resources/Scripts/ConvertFromModelica_LinearSystems2_2.3.5.mos [hunk 6]
```diff
@@ -0,0 +1,4 @@
+// Generated by conversion of Modelica_LinearSystems2
+convertElement("Modelica_LinearSystems2.Utilities.Plants.DoublePendulum","world.mue","world.mu");
+convertModifiers("Modelica_LinearSystems2.Utilities.Plants.DoublePendulumInverse",{"torque.torqueDiameter"},{"torque.connectionLineDiameter=%torque.torqueDiameter%"});
+convertElement("Modelica_LinearSystems2.Utilities.Plants.DoublePendulumInverse","world.mue","world.mu");
```
Comment (tobolar) [inline Modelica_LinearSystems2/Resources/Scripts/ConvertFromModelica_LinearSystems2_2.3.5.mos:4]: Covered by MSL conversion already?

Comment (HansOlsson) [reply Modelica_LinearSystems2/Resources/Scripts/ConvertFromModelica_LinearSystems2_2.3.5.mos:4]: Best to have duplicated.

**Pull request merged**